libs/blocks/bulk/utils.js

```diff
@@ -0,0 +1,267 @@
+import { getImsToken } from '../../../tools/send-to-caas/send-utils.js';
+import { fetchWithTimeout } from '../caas/utils.js';
+import { loadScript } from '../../utils/utils.js';
+import {
+  getLocalStorage,
+  setLocalStorage,
+} from '../review/utils/localStorageUtils.js';
+
+export const ADMIN_BASE_URL = 'https://admin.hlx.page';
+const THROTTLING_DELAY_MS = 100;
+export const BULK_CONFIG_FILE_PATH = '/tools/bulk-config.json';
+export const BULK_REPORT_FILE_PATH = '/tools/bulk-report';
+const BULK_AUTHORIZED_USERS = 'bulkAuthorizedUsers';
+export const BULK_SUPPORTED_SITES = 'bulkSupportedSites';
+export const BULK_STORED_URL_IDX = 'bulkStoredUrlIdx';
+const BULK_STORED_COMPLETED = 'bulkStoredCompleted';
+const BULK_STORED_URLS = 'bulkStoredUrls';
+const BULK_STORED_RESULTS = 'bulkStoredResults';
+const BULK_STORED_OPERATION = 'bulkStoredOperation';
+const UNSUPPORTED_SITE_STATUS = 'unsupported domain';
+const DUPLICATE_STATUS = 'duplicate';
+
+// eslint-disable-next-line no-promise-executor-return
+const delay = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+export const getStoredOperation = () => {
+  const name = getLocalStorage(BULK_STORED_OPERATION);
+  const urlIdx = getLocalStorage(BULK_STORED_URL_IDX);
+  const urls = getLocalStorage(BULK_STORED_URLS);
+  const completed = getLocalStorage(BULK_STORED_COMPLETED);
+  return {
+    name,
+    urlIdx,
+    urls,
+    completed,
+  };
+};
+
+export const getStoredUrlInput = () => {
+  const storedUrls = getStoredOperation().urls;
+  return storedUrls ? storedUrls.join('\n') : '';
+};
+
+export const storeUrls = (urls) => {
+  setLocalStorage(BULK_STORED_URLS, urls);
+};
+
+export const storeOperation = (operation) => {
+  setLocalStorage(BULK_STORED_OPERATION, operation);
+};
+export const getUser = async () => {
+  const profile = await window.adobeIMS?.getProfile();
+  return profile
+    ? { name: profile.name, email: profile.email }
+    : { name: 'anonymous', email: 'anonymous' };
+};
+
+export const signIn = async () => {
+  // force user to sign in
+  const accessToken = await getImsToken(loadScript);
+  if (!accessToken) {
+    window.adobeIMS.signIn();
+    return false;
+  }
+  return true;
+};
+
+export const signOut = (e) => {
+  e.preventDefault();
+  window.adobeIMS?.signOut();
+};
+
+export const getAuthorizedUsers = async () => {
+  let authorizedUsers = getLocalStorage(BULK_AUTHORIZED_USERS);
```

Comment:
Should there be some expiration on pulling from localstorage?

Reply:
@chrischrischris makes senses: the cache for authorized users and supported sites is cleared once the bulk operation is done: see 
https://github.com/adobecom/milo/pull/559/commits/d0c655477c907aa4565e2b3994d5066ba2ee59a5